### ISSUE TITLE
DB migration for fork choice cleanup

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#4233]
## Description

In #3962 we implemented some spec changes which simplified fork choice. As described in the comment of that issue, I created some DB migration debt. We also had some existing DB migration debt and with soaring interest rates it's never been a better time to pay it off.

There are two debts to pay off:

1. Remove the `best_justified_checkpoint` (debt from #3962)
1. Remove the `Option`s around the justified and finalized checkpoints on `ProtoNode` (existing debt introduced in #2822)

Solving (1) is quite straight forward; we don't need the [`best_justified_checkpoint`](https://github.com/sigp/lighthouse/blob/693886b94176faa4cb450f024696cb69cda2fe58/beacon_node/beacon_chain/src/beacon_fork_choice_store.rs#L367) field any more so create a new `superstruct` variant of `PersistedForkChoiceStore` and create a migration that ignores that field on any earlier variants.

(2) is a little more involved, however not too bad. The [justified and finalized checkpoint](https://github.com/sigp/lighthouse/blob/693886b94176faa4cb450f024696cb69cda2fe58/consensus/proto_array/src/proto_array.rs#L88-L91) fields don't need to be an option anymore. I would delete the option and then follow the errors upstream. Since we introduced those options in #2822 (the version at that time was v2.0.1), I don't think there's any need for us to handle the migration. We can just refuse to migrate a database from a pre-merge version to a post-capella version. I think it's unreasonable for anyone to expect that migration to work. We will have to some simple legwork to figure out the DB version in the next release that followed #2822.

So, in summary I think the DB migration can be quite simple. It just needs to drop the `best_justified_checkpoint` field and refuse to migrate from the DB version that was released with #2822.

**Comment by paulhauner** [comment]
Tagging @macladson since he showed some interest in this ☺️

**Comment by paulhauner** [comment]
Resolved via #4265 🎉
